### PR TITLE
Fe/in protected route

### PR DESCRIPTION
**Pull request by gagandeep95**
### What this PR does (required):
- It has a protected route that will only get access when the user is logged in. otherwise, it will only display login/signup pages.

### Any information needed to test this feature (required):
- user can enter the **localhost:3000/dashboard**  in the address bar to test this feature when it is not logged in

### CODE REVIEW

File: client/src/components/ProtectedRoute/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useAuth } from '../../context/useAuthContext';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface ProtectedRouteProps extends RouteProps {}
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute/ProtectedRoute.tsx:5]:
```suggestion
interface ProtectedRouteProps extends RouteProps {
authenticationPath: string;
exact: boolean
}
```

File: client/src/App.tsx [hunk 2]
```diff
@@ -23,20 +24,11 @@ function App(): JSX.Element {
...
-                </Route>
-                <Route exact path="/messages">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/my-sitters">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/edit-profile" component={EditMenu} />
-
+                <ProtectedRoute exact path="/dashboard" component={Dashboard} />
```
Comment (rajivtitus) [inline client/src/App.tsx:27]: 👍🏽

**Review by rajivtitus — APPROVED** [review]
Good work! Just address pending comment prior to merging, thanks! :)

Note: For future PRs please post screenshots/videos to show the feature in action.

File: client/src/components/ProtectedRoute/ProtectedRoute.tsx (hunk 1 above)
Comment (gagandeep95) [reply client/src/components/ProtectedRoute/ProtectedRoute.tsx:5]: Oh thanks, I will definitely add the types here.

**Pull request merged**